### ISSUE TITLE
About validation set

### ISSUE THREAD

**Issue by zeehuilin** [you359/Keras-FasterRCNN#2]
I see you divide the data set into three parts, but you don't seem to be using the validation set during training.

**Comment by you359** [comment]
actually, this repo cloned from https://github.com/yhenon/keras-frcnn/.
so, I'm not implement first version of this code from scratch.

Actually, i also found this issue, so i was trying to implement validation process.
However, Faster R-CNN training process in this repo is starting from fire directory something like flow_from_generator()
for example, when training the Faster R-CNN using train_frcnn.py, 
this script first parse Pascal VOC like annotation, (actually some another process also exist...) load image from parsed image path, and also extract ROIs(Bounding boxes) from annotation.
and then training Faster R-CNN using image, bbox information extracted above process.
this series of process is really time consuming.

for validate Object Detection Method, it need to calculate mean Average Precision.
you can explore mAP measuring process from measure_map.py
this process also time consuming.

So, Object Detection API such as Faster R-CNN implemented in https://github.com/tensorflow/models/tree/master/research/object_detection
is trained on TFRecord data type

To sum it up,
I was trying to implement validation process but it is actually difficult to implement with loading data from image file

**Comment by zeehuilin** [comment]
Thank you very much for your answer.
You've added a lot of BN (BatchNormalization) layers to xception.py.
In fact, I think there may be some problems with BN in Keras.I made a binary classification with your xaudience.py, but loss remained at 0.693 and did not converge. I also observed that in resnet50.py, the BN layer was fixedbatchnormalized.
In keras faster_rcnn (xception), RPN rpn_regr's loss declined slowly and detector_cls's loss declined quickly, which is quite different from keras faster_rcnn (vgg16) and keras faster_rcnn (res50).
Thank you again.

**Comment by latofa** [comment]
i have question if someone could help
i am beginner ,and i  training my data using Faster RCNN, but as mentioned the validation set code  in 'train_frcnn.py' is not used, so don't know how to create validation set for training my data with this situation.
where can i add the validation code, and how?
